api/test/test_helper/fake_android_plugin.gd

```diff
@@ -65,3 +98,34 @@ func getPackLocation(pack_name : String):
 	if pack_name in _asset_pack_location_store:
 		return _asset_pack_location_store[pack_name]
 	return null
+
+func getPackLocations():
+	return _asset_pack_location_store
+
+# -----------------------------------------------------------------------------
+# Simulates the cancel() function in PlayAssetDelivery Android plugin. Iterate
+# through all AssetPackState within _asset_pack_states_store and return the
+# updated states.
+# -----------------------------------------------------------------------------
+func cancel(pack_names : Array):
+	var return_asset_pack_states = {
+		PlayAssetPackStates._TOTAL_BYTES_KEY: 0,
+		PlayAssetPackStates._PACK_STATES_KEY: {}
+	}
+	
+	# iterate through all pack_names, if they exist in _asset_pack_states_store, try cancel them
+	for pack_name in pack_names:
+		if pack_name in _asset_pack_states_store[PlayAssetPackStates._PACK_STATES_KEY]:
```

Comment:
You could reduce the indentation in this loop by changing this to:
````
for pack_name in pack_names:
		var current_asset_pack_dict = _asset_pack_states_store[PlayAssetPackStates._PACK_STATES_KEY]:
                if current_asset_pack_dict == null:
                    continue;

                // per loop code.
````

Reply:
I think the ```if pack_name in _asset_pack_states_store[PlayAssetPackStates._PACK_STATES_KEY]:``` statement is required since GDScript will run into an ```Invalid get index``` error if looking up a non-existing key in Dictionary. I refactored this line so some variables can be reused.